test/distributed/_tensor/test_device_mesh.py

```diff
@@ -157,6 +157,27 @@ def test_device_mesh_hash(self):
         self.assertNotEqual(hash(mesh2), hash(mesh3))
 
 
+class InitDeviceMeshTest(DTensorTestBase):
+    @property
+    def world_size(self):
+        return 8
+
+    @with_comms
+    def test_init_device_mesh_with_mesh_dim_names(self):
+        mesh_shape = [2, 4]
+        ref_mesh = DeviceMesh(self.device_type, torch.arange(8).view(mesh_shape))
+
+        # test init_device_mesh with mesh_dim_names
+        mesh_dim_names = ["DP", "TP"]
```

Comment:
This should be a tuple instead? mypy probably won't complain as this is a test, but let's make it align with the init_device_mesh signature.